### PR TITLE
[state_store] delete get_value_at_version

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

we will use get_value_by_version instead at all callsites.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

ut

### CODE REVIEW

File: storage/aptosdb/src/state_store/mod.rs [hunk 1]
```diff
@@ -204,13 +204,20 @@ impl StateStore {
...
-        self.ledger_db
-            .get::<StateValueSchema>(key_and_version)
-            .and_then(|v| {
-                v.ok_or_else(|| format_err!("State Value is missing for {:?}", *key_and_version))
+    fn get_existing_value_by_version(
+        &self,
+        state_key: &StateKey,
+        version: Version,
+    ) -> Result<StateValue> {
+        self.get_value_by_version(state_key, version)
```
Comment (sitalkedia) [inline storage/aptosdb/src/state_store/mod.rs:212]: Can you clarify what is the motivation behind this change? My understanding is that `get_value_by_version` will be slower than `db.get` because of the seek - or is that not the case anymore?

Comment (lightmark) [reply storage/aptosdb/src/state_store/mod.rs:212]: @grao1991 is benchmarking this and the preliminary result is the `seek` is a little bit slower than `get` only but much faster than `get_with_proof` so we can sync tmr about the details.

We definitely need the `seek` version because lazy commit depends on this and we haven't found any workaround. If `prefix seek` can be tuned to be comparable to `get` we are good to go this route.

File: storage/aptosdb/src/state_store/mod.rs [hunk 2]
```diff
@@ -141,7 +141,7 @@ impl StateStore {
             JellyfishMerkleTree::new(self).get_with_proof(state_key.hash(), version)?;
         Ok((
             match leaf_data {
-                Some(x) => Some(self.get_value_at_version(&x.1)?),
+                Some(x) => Some(self.get_existing_value_by_version(&x.1 .0, x.1 .1)?),
```
Comment (msmouse) [inline storage/aptosdb/src/state_store/mod.rs:144]: too ugly.. give them names.

File: storage/aptosdb/src/state_store/mod.rs [hunk 3]
```diff
@@ -360,9 +367,9 @@ impl StateStore {
         Ok(
             JellyfishMerkleIterator::new(Arc::clone(self), version, start_hashed_key)?.map(
                 move |res| match res {
-                    Ok((_hashed_key, key_and_version)) => Ok((
-                        key_and_version.0.clone(),
-                        store.get_value_at_version(&key_and_version)?,
+                    Ok((_hashed_key, (key, version))) => Ok((
```
Comment (msmouse) [inline storage/aptosdb/src/state_store/mod.rs:370]: beautiful that they have names

File: storage/aptosdb/src/state_store/mod.rs [hunk 4]
```diff
@@ -204,13 +204,20 @@ impl StateStore {
...
     }
 
-    /// Read the value for a specific version and require its existence. Call `get_value_by_value`
-    /// for most use cases.
-    fn get_value_at_version(&self, key_and_version: &(StateKey, Version)) -> Result<StateValue> {
-        self.ledger_db
-            .get::<StateValueSchema>(key_and_version)
-            .and_then(|v| {
-                v.ok_or_else(|| format_err!("State Value is missing for {:?}", *key_and_version))
+    fn get_existing_value_by_version(
```
Comment (msmouse) [inline storage/aptosdb/src/state_store/mod.rs:207]: nit: name: "expect_value_by_version()"

**Review by msmouse — APPROVED** [review]
next step is removing "get_value_with_proof_by_version" altogether?

File: storage/aptosdb/src/state_store/mod.rs (hunk 2 above)
Comment (lightmark) [reply storage/aptosdb/src/state_store/mod.rs:144]: they are tuple. how do you assign names to tuple?hhh
